Source/OxyPlot/Rendering/TextArranger.cs

```diff
@@ -0,0 +1,259 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="TextArranger.cs" company="OxyPlot">
+//   Copyright (c) 2020 OxyPlot contributors
+// </copyright>
+// <summary>
+//   Arranges text.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace OxyPlot.Rendering
+{
+    using System;
+    using System.Linq;
+    using System.Xml.Linq;
+
+    /// <summary>
+    /// Arranges text.
+    /// </summary>
+    public class TextArranger
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TextArranger" /> class.
+        /// </summary>
+        /// <param name="textMeasurer">The <see cref="ITextMeasurer"/> to be used by this instance.</param>
+        /// <param name="textTrimmer">The <see cref="ITextTrimmer"/> to be used by this instance.</param>
+        public TextArranger(ITextMeasurer textMeasurer, ITextTrimmer textTrimmer)
+        {
+            this.TextMeasurer = textMeasurer ?? throw new ArgumentNullException(nameof(textMeasurer));
+            this.TextTrimmer = textTrimmer ?? throw new ArgumentNullException(nameof(textTrimmer));
+
+            this.SquashTrimmedTextBounds = false;
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ITextMeasurer"/> used by this instance.
+        /// </summary>
+        public ITextMeasurer TextMeasurer { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ITextTrimmer"/> used by this instance.
+        /// </summary>
+        public ITextTrimmer TextTrimmer { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to squash the bounds of trimmed text to the size the text.
+        /// </summary>
+        public bool SquashTrimmedTextBounds { get; set; }
```

Comment:
I can't quite figure out what exactly this property does - could this be explained a bit more clearly?

Reply:
I'm struggling to explain this clearly... maybe a picture will help for your benefit (obviously the documentation needs text).

This is `SquashTrimmedTextBounds = true`:
![squash](https://user-images.githubusercontent.com/5646475/91632345-0c7dcc00-e9d8-11ea-8cda-d24441daf8d4.PNG)

This is `SquashTrimmedTextBounds = false`:
![nosquash](https://user-images.githubusercontent.com/5646475/91632347-143d7080-e9d8-11ea-9fc0-1ff1be42f300.PNG)

When `true`, if text is trimmed horizontally, then the bounds are squashed to the bounds of trimmed value; otherwise they are 'clamped' to the max size. This only changes the way that the text is arranged, and looking at those two figures now, I think it should be `true`. We could remove it and just treat it as `true`... that would solve all my problems

Reply:
Just realised this only applies vertically... the horizontal is already bounded... I think I'll make it 'hard coded' `true` instead, to reduce confusion.